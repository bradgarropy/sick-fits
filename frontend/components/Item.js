import React from "react"
import PropTypes from "prop-types"
import Link from "next/link"
import format from "../utils/money"
import DeleteItem from "./DeleteItem"
import {ItemWrapper, ItemTitle, ItemButtons, PriceTag} from "../styles"

const Item = ({item}) => {
    const {id, title, price, description, image} = item

    return (
        <ItemWrapper>
            {image && <img src={image} alt={title}/>}

            <ItemTitle>
                <Link href={`/item/${id}`}>
                    <a>{title}</a>
                </Link>
            </ItemTitle>

            <PriceTag>{format(price)}</PriceTag>

            <p>{description}</p>

            <ItemButtons>
                <Link href={`/update/${id}`}>
                    <a>Edit ✍🏼</a>
                </Link>

                <button>Add To Cart 🛒</button>
                <DeleteItem id={id}>Delete ❌</DeleteItem>
            </ItemButtons>
        </ItemWrapper>
    )
}

Item.propTypes = {
    item: PropTypes.object.isRequired,
}

export default Item
