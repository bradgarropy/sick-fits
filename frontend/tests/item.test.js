import React from "react"
import {render} from "@testing-library/react"
import Item from "../components/Item"

const item = {
    id: "abc123",
    title: "cool item",
    price: 5000,
    description: "this item is really cool!",
    image: "dog.jpg",
    largeImage: "large-dog.jpg",
}

describe("<Item>", () => {
    test("snapshot", () => {
        const {container} = render(<Item item={item}/>)

        expect(container).toMatchSnapshot()
    })

    test("title", () => {
        const {getByText} = render(<Item item={item}/>)
        const title = getByText(item.title)

        expect(title.tagName).toEqual("A")
    })

    test("price", () => {
        const {getByText} = render(<Item item={item}/>)
        const price = getByText("$50")

        expect(price.tagName).toEqual("SPAN")
    })

    test("image", () => {
        const {getByAltText} = render(<Item item={item}/>)
        const image = getByAltText(item.title)

        expect(image.tagName).toEqual("IMG")
        expect(image.attributes.src.value).toEqual(item.image)
    })

    test("buttons", () => {
        const {getByText} = render(<Item item={item}/>)
        const editButton = getByText(/Edit/)
        const addToCartButton = getByText(/Add To Cart/)
        const deleteButton = getByText(/Delete/)

        expect(editButton.tagName).toEqual("A")
        expect(addToCartButton.tagName).toEqual("BUTTON")
        expect(deleteButton.tagName).toEqual("BUTTON")
    })
})
