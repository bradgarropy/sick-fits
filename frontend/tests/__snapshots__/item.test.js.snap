// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item> snapshot 1`] = `
<div>
  <div
    class="ItemWrapper-sc-1q31uz2-0 eZrZzJ"
  >
    <img
      alt="cool item"
      src="dog.jpg"
    />
    <h3
      class="ItemTitle-sc-9qzhwq-0 hjyVUg"
    >
      <a
        href="/item/abc123"
      >
        cool item
      </a>
    </h3>
    <span
      class="PriceTag-sc-1yto7zk-0 fvHRHo"
    >
      $50
    </span>
    <p>
      this item is really cool!
    </p>
    <div
      class="ItemButtons-adym8q-0 fjwJcq"
    >
      <a
        href="/update/abc123"
      >
        Edit ✍🏼
      </a>
      <button>
        Add
         To Cart 🛒
      </button>
      <button>
        Delete ❌
      </button>
    </div>
  </div>
</div>
`;
